craft_cli/messages.py

```diff
@@ -242,6 +242,16 @@ def message(self, text: str, intermediate: bool = False) -> None:
         )
         self.printer.show(sys.stdout, text, use_timestamp=use_timestamp)  # type: ignore
 
+    @_init_guard
+    def trace(self, text: str) -> None:
+        """Trace/debug information.
+
+        This is to record everything that the user may not want to normally see, but it's
+        useful for postmortem analysis.
+        """
+        stream = sys.stderr if self.mode == EmitterMode.TRACE else None
```

Comment:
This might solve your issue
```suggestion
        if self.mode == EmitterMode.TRACE:
            stream: Optional[typing.TextIO] = sys.stderr
        else:
            stream = None
```

Reply:
Nah, I like it the way it ended up.

Reply:
Thanks!